### ISSUE TITLE
TypeError: Cannot read property 'isSingle' of undefined

### ISSUE THREAD

**Issue by crapthings** [cult-of-coders/grapher#108]
linker order matter

this will cause
TypeError: Cannot read property 'isSingle' of undefined

```
Contracts.addLinks({
  orders: {
    collection: Orders,
    inversedBy: 'contract',
  }
})

Orders.addLinks({
  contract: {
    type: 'one',
    collection: Contracts,
    field: 'contractId',
    index: true,
  }
})
```

work one

```
Orders.addLinks({
  contract: {
    type: 'one',
    collection: Contracts,
    field: 'contractId',
    index: true,
  }
})

Contracts.addLinks({
  orders: {
    collection: Orders,
    inversedBy: 'contract',
  }
})
```

what if we can scan all define before addlinks

**Comment by theodorDiaconu** [comment]
@crapthings we did something that should have fixed this. in the latest version, pls confirm you're on latest so I can investigate.

**Comment by crapthings** [comment]
this branch is an reproduce

i am on 1.2.6

https://github.com/crapthings/grapher-test/blob/nested-reduce-reuse/lib/collections.js

**Comment by theodorDiaconu** [comment]
@crapthings just tried your reproduction. For me it worked, I got no errors.

**Comment by theodorDiaconu** [comment]
Pls try it again and triple-check you are on 1.2.6, because we solved the issue exactly for that in 1.2.6
